lib/Workflow/Condition.pm

```diff
@@ -25,6 +29,140 @@ sub evaluate {
     croak "Class ", ref($self), " must implement 'evaluate()'!\n";
 }
 
+
+sub evaluate_condition {
+    my ( $class, $wf, $condition_name) = @_;
+    $log ||= get_logger();
+    $wf->type;
+
+    my $factory = $wf->_factory();
+    my $orig_condition = $condition_name;
+    my $opposite       = 0;
+    my $condition;
+
+    $log->is_debug
+        && $log->debug("Checking condition $condition_name");
+
+    if ( $condition_name =~ m{ \A ! }xms ) {
+
+        # this condition starts with a '!' and is thus supposed
+        # to return the opposite of an original condition, whose
+        # name is the same except for the '!'
+        $orig_condition =~ s{ \A ! }{}xms;
+        $opposite = 1;
+        $log->is_debug
+            && $log->debug(
+            "Condition starts with a !: '$condition_name'");
+    }
+
+    local $wf->{'_condition_result_cache'} =
+        $wf->{'_condition_result_cache'} || {};
+    if ( $Workflow::Condition::CACHE_RESULTS
+         && exists $wf->{'_condition_result_cache'}->{$orig_condition} ) {
+
+        # The condition has already been evaluated and the result
+        # has been cached
+        $log->is_debug
+            && $log->debug(
+            "Condition has been cached: '$orig_condition', cached result: ",
+            $wf->{'_condition_result_cache'}->{$orig_condition}
+            );
+        if ( !$opposite ) {
+            $log->is_debug
+                && $log->debug("Opposite is false.");
```

Comment:
I find the term "opposite" confusing, do you mean inverted?

Reply:
Negated was even better.